### PR TITLE
Commit Gradle Wrapper files into git

### LINKED ISSUE #127891: Flutter Gradle Plugin is missing the Gradle Wrapper
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Make sure you're on Flutter's latest `master` channel
1. Make sure that `git status` executed in Flutter's directory shows no changes 
1. Go to the directory where Flutter is installed, and then go to `examples/hello_world`
2. Run `flutter build apk --debug` in `examples/hello_world`
3. Run `git status` again, and see that Gradle files were generated:

### Expected results

`git status` should show no changes at all.

In other words, the files that are generated should be committed to git.

### Actual results

```
$ git status
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	../../packages/flutter_tools/gradle/gradle/
	../../packages/flutter_tools/gradle/gradlew
	../../packages/flutter_tools/gradle/gradlew.bat

nothing added to commit but untracked files present (use "git add" to track)
```

### What is going on?

Before #123511, the Flutter Gradle Plugin was built using the app's project Gradle and applied using the imperative `apply` function.

After changes made in #123511, Flutter Gradle Plugin is now also a "normal", standalone Gradle plugin project, that _can_ be included in the app's Gradle project using `includeBuild()`. Please note the _can_ - most apps still use the old way of using the FGP, so they do not experience this problem.. But all newly `flutter create`d apps will experience this.

Committing an unverified Gradle Wrapper into a repository may pose a security threat. There are ways to protect the wrapper against supply chain attacks:
- [Protecting Project Integrity | Gradle blog post](https://blog.gradle.org/project-integrity)
- [Verifying the integrity of the Gradle Wrapper JAR | Gradle docs](https://docs.gradle.org/8.1/userguide/gradle_wrapper.html#wrapper_checksum_verification)

TL;DR IMHO the most sensible solution is to use the [`gradle/gradle-wrapper-validation`](https://github.com/marketplace/actions/gradle-wrapper-validation) GitHub Action.

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
$ flutter doctor --verbose
[✓] Flutter (Channel master, 3.11.0-16.0.pre.11, on macOS 13.3.1 22E772610a darwin-arm64, locale en-PL)
    • Flutter version 3.11.0-16.0.pre.11 on channel master at /Users/bartek/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 1dc26f80f0 (3 hours ago), 2023-05-30 15:37:12 -0400
    • Engine revision a6d3124ced
    • Dart version 3.1.0 (build 3.1.0-155.0.dev)
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/bartek/androidsdk
    • Platform android-33, build-tools 33.0.1
    • ANDROID_HOME = /Users/bartek/androidsdk
    • Java binary at: /Users/bartek/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.10121639/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Users/bartek/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.10121639/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.2)
    • IntelliJ at /Users/bartek/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin version 73.1.1
    • Dart plugin version 231.9065

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.2)
    • IntelliJ at /Users/bartek/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/231.9011.34/IntelliJ
      IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.78.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.3.1 22E772610a darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 113.0.5672.126

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by bartekpacia**
This PR fixes #127891.

These files are freshly generated. But of course, do not trust me – verify them yourself ([guide on how to do it](https://docs.gradle.org/8.1/userguide/gradle_wrapper.html#wrapper_checksum_verification)).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by bartekpacia** [comment]
Oops, looks like adding a binary blob to this repo is not as easy as I thought :)
